src/System.Net.Http/src/System/Net/Http/Managed/HttpEnvironmentProxy.cs

```diff
@@ -0,0 +1,199 @@
+
+            return new HttpEnvironmentProxy(http, https);
+        }
+
+        private HttpEnvironmentProxy(string http, string https)
+        {
+
+            if (!String.IsNullOrWhiteSpace(http))
+            {
+                _http = GetUriFromString(http);
+                if (!string.IsNullOrWhiteSpace(_http.UserInfo))
+                {
+                    String[] s = _http.UserInfo.Split(':', 2);
+                    _httpCred = new NetworkCredential(s[0], s[1]);
+                }
+            }
+            if (!String.IsNullOrWhiteSpace(https))
+            {
+                _https = GetUriFromString(https);
+                if (!string.IsNullOrWhiteSpace(_https.UserInfo))
+                {
+                    String[] s = _https.UserInfo.Split(':', 2);
+                    _httpsCred = new NetworkCredential(s[0], s[1]);
+                }
+            }
+
+            string value = Environment.GetEnvironmentVariable("no_proxy");
+            if (!String.IsNullOrWhiteSpace(value)) 
+            {
+                _bypass = value.Split(',');
+            }
+        }
+
+        /// <summary>
+        /// This function will evaluate given string and it will try to convert
+        /// it to Uri object. The string could contain URI fragment, IP address and  port
+        /// tuple or just IP address or name. 
+        /// </summary>
+        private Uri GetUriFromString(String value)
+        {
+            if (!value.StartsWith("http://", StringComparison.OrdinalIgnoreCase) &&
+                !value.StartsWith("https://", StringComparison.OrdinalIgnoreCase))
+            {
+                // Default to HTTP
+                value = "http://" + value;
+            }
+            Uri uri = new Uri(value);
+
+            if (uri.Scheme != Uri.UriSchemeHttp && uri.Scheme != Uri.UriSchemeHttps)
+            {
+                throw new NotSupportedException("Only HTTP and HTTPS protocols are supported for proxy.");
+            }
+            return uri;
+        }
+
+        /// <summary>
+        /// This function returns true if given Host mach bypss list.
+        /// Note, that the list is common for http and https.
+        /// </summary>
+        private bool IsMatchInBypassList(Uri input)
+        {
+            if (_bypass != null)
+            {
+                foreach (string s in _bypass)
+                {
+                    if (s[0] == '.')
+                    {
+                        // This should match either domain it self or any subdomain or host
+                        // .foo.com will match foo.com it self or *.foo.com
+                        if ((s.Length - 1) == input.Host.Length &&
+                            String.Compare(s, 1, input.Host, 0, input.Host.Length, StringComparison.OrdinalIgnoreCase) == 0)
+                        {
+                            return true;
+                        }
+                        else if (input.Host.EndsWith(s, StringComparison.OrdinalIgnoreCase))
```

Comment:
So if my host is microsoft.com and I bypass soft.com, we still want that to be bypassed?

Reply:
Ah, nevermind, this is only done when it begins with a period.